src/neynarAPI/utils.ts

```diff
@@ -0,0 +1,42 @@
+import { mnemonicToAccount } from "viem/accounts";
+
+export async function generateSignature(
+  publicKey: string,
+  appFid: number,
+  accountMnemonic: string,
+  deadline: number
+): Promise<string> {
+  // DO NOT CHANGE ANY VALUES IN THIS CONSTANT
+  const SIGNED_KEY_REQUEST_VALIDATOR_EIP_712_DOMAIN = {
+    name: "Farcaster SignedKeyRequestValidator",
+    version: "1",
+    chainId: 10,
+    verifyingContract:
+      `0x00000000fc700472606ed4fa22623acf62c60553` as `0x${string}`,
```

Comment:
Huh. Do you know what that type assertion is doing? I've never seen a type assertion combined with a string template that way

Reply:
This was the only way I could get it to compile, it kept complaining that
```
Type 'string' is not assignable to type '`0x${string}`'.

```
so I defined it explicitly.

The code was taken from [here](https://github.com/neynarxyz/farcaster-signed-key/blob/main/generate_signature.js)